0-common/schema-ddl/src/test/scala/com/snowplowanalytics/iglu/schemaddl/jsonschema/StringSpec.scala

```diff
@@ -35,25 +36,42 @@ class StringSpec extends Specification with ValidationMatchers { def is = s2"""
       """
         |{"minLength": 32}
       """.stripMargin)
-
-    Schema.parse(schema) must beSome(Schema(minLength = Some(MinLength(32))))
+    val formats = DefaultFormats + MinLengthSerializer
+    val mf = implicitly[Manifest[MinLength]]
+    val result = schema.extractOpt[MinLength](formats, mf)
+    result must beSome(MinLength(32))
   }
 
   def e2 = {
     val schema = parse(
       """
         |{"maxLength": 32, "format": "ipv4"}
       """.stripMargin)
+    val formats = DefaultFormats + MaxLengthSerializer + FormatSerializer
+    val mf1 = implicitly[Manifest[MaxLength]]
+    val result1 = schema.extractOpt[MaxLength](formats, mf1)
+    result1 must beSome(MaxLength(32))
 
-    Schema.parse(schema) must beSome(Schema(maxLength = Some(MaxLength(32)), format = Some(Ipv4Format)))
+    val mf2 = implicitly[Manifest[Ipv4Format.type]]
+    val result2 = schema.extractOpt[Ipv4Format.type](formats, mf2)
+    result2 must beSome(Ipv4Format)
   }
 
   def e3 = {
     val schema = parse(
       """
         |{"maxLength": 32, "format": "unknown"}
       """.stripMargin)
+    println(schema)
+    val formats = DefaultFormats + MaxLengthSerializer + FormatSerializer
+    val mf1 = implicitly[Manifest[MaxLength]]
+    val result1 = schema.extractOpt[MaxLength](formats, mf1)
+    println(result1)
+    result1 must beSome(MaxLength(32))
 
-    Schema.parse(schema) must beSome(Schema(maxLength = Some(MaxLength(32)), format = Some(CustomFormat("unknown"))))
+    val mf2 = implicitly[Manifest[CustomFormat]]
```

Comment:
Also, we don't have serializer for `CustomFormat`, only for `Format` which is root of sealed (ADT) hierarchy. Thus you need to extract `Format` here.